ui/utils.py

```diff
@@ -0,0 +1,18 @@
+from functools import wraps
+
+
+def to_be_foreground(func):
+    # A safety check wrapper so that certain functions can't possibly be called
+    # if UI element is not the one active
+    @wraps(func)
+    def wrapper(self, *args, **kwargs):
+        if self.in_foreground:
+            return func(self, *args, **kwargs)
+        else:
+            return False
+
+    return wrapper
+
+
+def clamp(value, _min, _max):
```

Comment:
Great function! In base_list_ui.py, there's a `fix_pointers_on_contents_update` function that could make use of this, so if you have some spare time, could add it there (but do make sure to preserve comments where possible)

Reply:
Sure !

Reply:
Done